### ISSUE TITLE
[MU4 Issue] Stems cannot be colored

### ISSUE THREAD

**Issue by micpap25** [musescore/MuseScore#11159]
**Describe the bug**
Discovered while testing for #11096 and somewhat related.
If a stem is selected and then a color is chosen, the stem's color does not change and the notehead's color changes.
I believe this is because, as described in the linked issue, the entire chord is selected by the "General" tool and this is the way that MuseScore renders the note's color value.

**To Reproduce**
1. Create a new note.
2. Click the note's stem.
3. Go to **General > Appearance > Color**
4. Change the note's color in the Color Picker.
5. Click "OK"

**Expected behavior**
The color of the stem should change, and the color of the notehead should not change.

**Platform information**
 - OS: Windows 11

**Additional context**
This is not a problem on MU3

**Comment by Tantacrul** [comment]
Agree. Unlike some other parameters, colour should apply to the selected element only.

1. If a user selects a note head and changes the colour it should only affect the notehead
2. If a user selects a stem and changes the colour it should only affect the stem
<img width="52" alt="image" src="https://user-images.githubusercontent.com/21022311/166443080-5eefd1b6-0db8-4e1f-aa8c-ad36e80f70ba.png">


3. If a user range-selects multiple notes, and changes the colour, it'll affect notes, stems, flags and beams
<img width="395" alt="image" src="https://user-images.githubusercontent.com/21022311/166443175-23889349-89a4-4e65-a770-509884cb621e.png">

4. etc.

**Comment by wizofaus** [comment]
Happy to work on this if your proposal is definitely what you want to go with @Tantacrul.

**Comment by Tantacrul** [comment]
Yes please! Do tell me if there is anything stupid about my suggestion. I think it makes sense but there might be an edge case or two I haven't considered.

**Comment by wizofaus** [comment]
There's nothing stupid about it but I can't see a way of implementing this properly without some big changes.
Basically when you select a stem, the "inspector" considers only the chord that it's a part of.  For many properties this makes sense, but for others it doesn't, and there doesn't seem to be a good mechanism in place to allow treating different properties differently (such that for some, you go to the "chord" level, others to the note level and others to the stem/notehead level). 
Much as I hate to say it, changing stem colour independently seems a sufficiently niche feature that I'm not sure it's justifiable at this point in the release cycle.
However what *needs* to be fixed is that the stem colour isn't changed at all even if you select the whole chord (via range selection). So I'll focus on that.

**Comment by Tantacrul** [comment]
OK cool, let us know and I'll give it a quick test. Thanks!

**Comment by wizofaus** [comment]
Do you want me to treat the "offset" properties similarly?  Have you seen what happens currently when you select a range including a chord, and try to edit other values under "Appearance" (including offset)?

**Comment by Tantacrul** [comment]
AFAICS, the offset option works as expected for range and individual note selection (we intentionally changed it from V3). Basically, if you want to separate a notehead, there's a much more hidden offset option under 'Notehead', etc.

For now, I'd prefer to avoid bringing offset into this discussion just because it'll make it really hard for me to come back to.

Again, I can test the proposed solution and then figure out if I have any points to raise.

**Comment by wizofaus** [comment]
Sorry, took a while to get around to it, but have put up a PR now, literally all it does is treat "color" the same as "visibility" - i .e. when you apply it to a chord (via a range selection), it will apply to the stem etc. too. Though it doesn't seem to be applying to augmentation dots (which is odd because visibility does work there, suggesting there's magical logic elsewhere for dots to automatically take their visibility from the note they're attached to)

**Comment by wizofaus** [comment]
Ah, so visibility isn't working as expected either, if you change via the "visible" checkbox in the properties panel/inspector. Will fix that too.

**Comment by bkunda** [comment]
🟢 Re-tested in 4.0.2. Remains FIXED.
